### ISSUE TITLE
Delay between barcode scans & Camera frame rate while scanning

### ISSUE THREAD

**Issue by samigimix** [rmtmckenzie/flutter_qr_mobile_vision#37]
Thanks for create this useful package 👍
I have 2 problem and have no idea how to handle them : 

1. QrCamera's **callback** called too many times immediately after scanned barcode. I want to add a specific delay after each callback and QrCamera can't scan & callback until the end of delay.

2. QrCamera's **camera frame rate** is so low and I want to change it to normal mode.

**Comment by rmtmckenzie** [comment]
As to #1, I have been planning on eventually adding some logic that makes it only return a new QT code - that's better than a delay I believe. You could do the same in your callback fairly simply for now though.

The second issue is... complicated. Because of all the way the camera works particularly on Android I have to do some non-optimal things to support both camera API's and the QR processing at the same time. When I get some time I'm planning on doing some more tweaking as I realize performance is the biggest problem for this plugin (esp on Android, iOS is much smoother).

One thing you can do to improve performance without any significant code changes is to specify exactly which type of QR/Barcode you need scanned if you only need one type. That way there's less processing to do for each frame.

**Comment by samigimix** [comment]
As to #1 I used **Timer** and it handled easily.
As to #2 I just using **Barcode**, But where I should specify that ? mention in code ?
Or is there a sub package just for barcode in your package ?

**Comment by rmtmckenzie** [comment]
QrCamera has a parameter 'formats' which you can pass in, with the formats currently being:

```dart
enum BarcodeFormats {
  ALL_FORMATS,
  AZTEC,
  CODE_128,
  CODE_39,
  CODE_93,
  CODABAR,
  DATA_MATRIX,
  EAN_13,
  EAN_8,
  ITF,
  PDF417,
  QR_CODE,
  UPC_A,
  UPC_E,
}
```

**Comment by samigimix** [comment]
Thanks, It helped and performance improved 👍
But if you can write this package **natively** on iOS and android instead of dart, performance improved completely cause dart language is slow basically.
Anyway thanks for support and good luck.

**Comment by rmtmckenzie** [comment]
The camera processing is being done natively (well, java at least) for both android and iOS. And the QrCamera passes its size through to the native side so that it can choose the best size of camera input based on that size - so if your view is smaller or you have a lower resolution device it will choose a smaller image to process. Once (if) flutter supports c++ I will definitely rewrite this using that =D.

The resolution is something I could tweak although it would result in worse resolution for the view, but if it's smoother that might be a good compromise. I do still need to make sure the picture has enough resolution to actually find the QR code though - when I lowered it the QR was a lot harder to detect and I erred on the side of it working better rather than looking better.
